lib/fs.js

```diff
@@ -1384,6 +1384,33 @@ function mkdirSync(path, options) {
   }
 }
 
+function readdirSyncRecursive(origPath, options) {
+  function _readdirSyncRecursive(path) {
```

Comment:
I don't think we need both `readdirSyncRecursive()` and `_readdirSyncRecursive()`. You can let `readdir()` call `getValidatedPath()` and then call what is currently `_readdirSyncRecursive()` if you pass `options` to it.